src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -525,6 +612,115 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
             return _requestPathToResources;
         }
 
+        private string? GetDefaultNameFromConfiguration(OperationSet operationSet, ResourceTypeSegment resourceType)
+        {
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue(operationSet.RequestPath, out var name))
+                return name;
+            if (_context.Configuration.MgmtConfiguration.RequestPathToResourceName.TryGetValue($"{operationSet.RequestPath}|{resourceType}", out name))
+                return name;
+
+            return null;
+        }
+
+        private string GetResourceName(string candidateName, OperationSet operationSet, RequestPath requestPath)
+        {
+            // read configuration to see if we could get a configuration for this resource
+            var resourceType = requestPath.GetResourceType(_mgmtConfiguration);
+            var defaultNameFromConfig = GetDefaultNameFromConfiguration(operationSet, resourceType);
+            if (defaultNameFromConfig != null)
+                return defaultNameFromConfig;
+
+            var resourcesWithSameName = _resourceDataSchemaNameToOperationSets[candidateName];
+            var resourcesWithSameType = ResourceOperationSets
+                .SelectMany(opSet => opSet.GetRequestPath(_context).Expand(_mgmtConfiguration))
+                .Where(rqPath => rqPath.GetResourceType(_mgmtConfiguration).Equals(resourceType))
+                .ToArray();
```

Comment:
Looks like we are only iterating this collection, therefore I believe we do not need to convert this to an array.

Reply:
it made debugging a LOT easier but I can remove that now